python/cudf/cudf/utils/ioutils.py

```diff
@@ -2099,3 +2095,96 @@ def _read_byte_ranges(
 
     for worker in workers:
         worker.join()
+
+
+def _get_remote_bytes_all(
+    remote_paths, fs, *, blocksize=_BYTES_PER_THREAD_DEFAULT
+):
+    if (
+        len(remote_paths) >= 8  # Heuristic to avoid fs.sizes
```

Comment:
Is there an issue/PR/doc/something we can link to to explain what we're avoiding and why? If not, could we add a little longer comment (I assume this is some performance bottleneck)?

Reply:
I removed this for now and left a comment that we may want to explore a heuristic like this in the future.